ui/components-react/src/components-react/newEditors/CommittingEditor.tsx

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import * as React from "react";
+import { useRef, useState } from "react";
+import { Editor } from "./Editor.js";
+import type { Value } from "./values/Values.js";
+import type { EditorProps } from "./Types.js";
+
+type CommittingEditorProps = Omit<EditorProps, "onFinish" | "onChange"> & {
+  onCommit: (value: Value) => void;
+  onCancel?: () => void;
+};
+
+/**
+ *
+ */
+export function CommittingEditor({
+  metadata,
+  value,
+  onCommit,
+  onCancel,
+  size,
+  ...restProps
+}: CommittingEditorProps) {
+  const [currentValue, setCurrentValue] = useState<Value | undefined>();
+  const currentValueRef = useRef(currentValue);
+
+  const handleChange = (newValue: Value) => {
+    currentValueRef.current = newValue;
+    setCurrentValue(newValue);
+  };
+
+  const handleCommit = () => {
+    if (currentValueRef.current !== undefined) {
+      onCommit(currentValueRef.current);
+      return;
+    }
+
+    onCancel?.();
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    e.stopPropagation();
+
+    if (e.key === "Enter" || e.key === "Tab") {
+      handleCommit();
+    } else if (e.key === "Escape") {
+      onCancel?.();
+    }
+  };
+
+  return (
+    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
+    <div onKeyDown={handleKeyDown}>
+      <Editor
```

Comment:
Is `Editor` component expected to ever be used outside of here? Maybe it should be moved into this source file and not exported?

Reply:
Yes. I expect to use it in `FilterBuilder` component where we treat editor as a controlled input and do not wait for committing action.